Api.Abstractions/Extensions/ServiceExtensions.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Text;
+using Diagnosea.Submarine.Api.Abstractions.Settings;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.OpenApi.Models;
+using Swashbuckle.AspNetCore.Filters;
+
+namespace Diagnosea.Submarine.Api.Abstractions.Extensions
+{
+    public static class ServiceExtensions
+    {
+        public static void AddSubmarineAuthentication(this IServiceCollection serviceCollection, ISubmarineAuthenticationSettings settings)
+        {
+            serviceCollection.AddAuthentication(x =>
```

Comment:
Potentially could add some sort of mapper instead of this lambda, but that might be overengineering. Just wondering how we can abstract this.